test/models/trip_test.dart

```diff
@@ -0,0 +1,369 @@
+        hideInListView: true,
+        startTime: DateTime.now().add(const Duration(hours: 2)),
+        endTime: DateTime.now().add(const Duration(hours: 4)),
+      );
+      expect(drive.shouldShowInListView(past: false), false);
+    });
+    test('returns true for past finsihed trip', () {
+      final Trip ride = RideFactory().generateFake(
+        status: RideStatus.approved,
+        startTime: DateTime.now().subtract(const Duration(hours: 4)),
+        endTime: DateTime.now().subtract(const Duration(hours: 2)),
+      );
+      expect(ride.shouldShowInListView(past: true), true);
+      final Trip drive = DriveFactory().generateFake(
+        cancelled: false,
+        startTime: DateTime.now().subtract(const Duration(hours: 4)),
+        endTime: DateTime.now().subtract(const Duration(hours: 2)),
+      );
+      expect(drive.shouldShowInListView(past: true), true);
+    });
+    test('returns false in ongoing trip', () {
+      final Trip ride = RideFactory().generateFake(
+        status: RideStatus.pending,
+        startTime: DateTime.now().subtract(const Duration(hours: 2)),
+        endTime: DateTime.now().add(const Duration(hours: 4)),
+      );
+      expect(ride.shouldShowInListView(past: true), false);
+      final Trip drive = DriveFactory().generateFake(
+        cancelled: true,
+        startTime: DateTime.now().subtract(const Duration(hours: 2)),
+        endTime: DateTime.now().add(const Duration(hours: 4)),
+      );
+      expect(drive.shouldShowInListView(past: true), false);
+    });
+  });
+  group('Trip.shouldShowInListView', () {
+    test('shows in ListView', () {
+      final Trip trip = (randomTripFactory() as TripFactory).generateFake(
+        startTime: DateTime.now().add(const Duration(hours: 2)),
+        endTime: DateTime.now().add(const Duration(hours: 4)),
+      );
+      expect(trip.shouldShowInListView(past: false), true);
+    });
+    test('hideInListView', () {
+      final Trip trip = (randomTripFactory() as TripFactory).generateFake(
+        startTime: DateTime.now().add(const Duration(hours: 2)),
+        endTime: DateTime.now().add(const Duration(hours: 4)),
+        hideInListView: true,
+      );
+      expect(trip.shouldShowInListView(past: false), false);
+    });
+    test('shows in past ListView', () {
+      final Trip trip = (randomTripFactory() as TripFactory).generateFake(
+        startTime: DateTime.now().subtract(const Duration(hours: 4)),
+        endTime: DateTime.now().subtract(const Duration(hours: 2)),
+      );
+      expect(trip.shouldShowInListView(past: true), true);
+    });
+    test('not showing in ListView', () {
+      final Trip trip = (randomTripFactory() as TripFactory).generateFake(
+        startTime: DateTime.now().subtract(const Duration(hours: 4)),
+        endTime: DateTime.now().subtract(const Duration(hours: 2)),
+      );
+      expect(trip.shouldShowInListView(past: false), false);
+    });
+  });
+  group('Trip.equals', () {
```

Comment:
Am I wrong or does this not have a test case that is actually equal? :D

Reply:
Forgot to add that one here